src/mpid/ch4/generic/am/mpidig_am_recv.h

```diff
@@ -137,10 +137,9 @@ MPL_STATIC_INLINE_PREFIX int MPIDIG_do_irecv(void *buf, MPI_Aint count, MPI_Data
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDIG_DO_IRECV);
 
     root_comm = MPIDIG_context_id_to_comm(context_id);
-    unexp_req = MPIDIG_dequeue_unexp(rank, tag, context_id, &MPIDIG_COMM(root_comm, unexp_list));
+    unexp_req = MPIDIG_dequeue_unexp(rank, tag, context_id, NULL);
```

Comment:
How about pass the global list in? That way, `MPDIG_dequeue_unexp` gets closer to a utility and less reliant on the global context.

Reply:
Yes, my follow up PR adds that in another commit.

Reply:
https://github.com/pmodels/mpich/pull/5280/commits/18b31b7bba7ae024c4068e9400954f2c825b50f1